pkg/cloudprovider/cloudprovider.go

```diff
@@ -181,18 +199,40 @@ func kubeDNSIP(ctx context.Context, kubernetesInterface kubernetes.Interface) (n
 	return kubeDNSIP, nil
 }
 
-func (c *CloudProvider) resolveNodeTemplate(ctx context.Context, provider *runtime.RawExtension, providerRef *v1alpha5.ProviderRef) (*v1alpha1.AWSNodeTemplate, error) {
+func (c *CloudProvider) resolveNodeTemplate(ctx context.Context, raw []byte, objRef *v1.ObjectReference) (*v1alpha1.AWSNodeTemplate, error) {
```

Comment:
This lets you hide the annotation when it's removed later.
```suggestion
func (c *CloudProvider) resolveNodeTemplate(ctx context.Context, v1alpha5.Machine) (*v1alpha1.AWSNodeTemplate, error) {
```